edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+
+namespace Microsoft.Azure.Devices.Edge.Hub.Service
+{
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class CertificateRenewal : IDisposable
+    {
+        readonly CancellationTokenSource cts;
+
+        public CancellationToken Token => this.cts.Token;
+
+        public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger)
+        {
+            Preconditions.CheckNotNull(certificates, nameof(certificates));
+            Preconditions.CheckNotNull(logger, nameof(logger));
+
+            TimeSpan timeToExpire = certificates.ServerCertificate.NotAfter - DateTime.UtcNow;
+            if (timeToExpire > TimeSpan.Zero)
+            {
+                logger.LogInformation("Scheduling server certificate renewal for {0}.", DateTime.UtcNow.Add(timeToExpire).ToString("o"));
+                this.cts = new CancellationTokenSource(timeToExpire);
+                this.cts.Token.Register(l => ((ILogger)l).LogInformation("Performing server certificate renewal."), logger);
+            }
+            else
+            {
+                this.cts = new CancellationTokenSource();
+                logger.LogWarning("Time to server certificate expiration is {0}. Not scheduling renewal.", timeToExpire.ToString("c"));
```

Comment:
Should we instead say "Certificate is expired".. might be better?
Also, in this case we don't do anything?

Reply:
Yeah, I'm worried about it getting in a crash loop?

Reply:
Hmm, I see.. I was thinking purely form this class's standpoint.. 
About the crash loop, isn't that what will happen anyways? Will the servers be able to start properly?